warehouse/query-core/src/main/java/datawave/query/index/lookup/Union.java

```diff
@@ -311,4 +310,88 @@ public String getContextDebug() {
     public JexlNode currentNode() {
         return currNode;
     }
+    
+    /**
+     * Seek every IndexStream whose top shard is lower than the specified shard.
+     *
+     * @param seekShard
+     *            the shard to seek to.
+     * @return the lowest shard within this union after seeking
+     */
+    @Override
+    public String seek(String seekShard) {
+        
+        // Guard against the case when a seek range like YYYYMMDD_ is provided.
+        if (seekShard.charAt(seekShard.length() - 1) == '_')
+            seekShard = new String(seekShard.getBytes(), 0, seekShard.length() - 1);
+        
+        // Check the current element first
+        String topShard = isTopElementAMatch(seekShard);
+        if (topShard != null) {
+            // If the top element is a day and we are seeking to a shard range within the day, re-map the top element to the shard range.
+            // Take (A_shard && (B_day || C_day)) for example. The union's top shard is a day, but must be mapped to the same shard as the A term in order for
+            // the intersection to recognize that an intersection is possible.
+            if (!isDay(seekShard) && ShardEquality.matches(this.next.first(), seekShard)) {
+                this.next = new Tuple2<>(seekShard, this.next.second());
+            }
+            
+            return topShard;
+        } else {
+            // If the top element did not match then null it out
+            this.next = null;
+        }
+        
+        List<IndexStream> nextChildren = new ArrayList<>();
+        while (!children.isEmpty()) {
+            
+            IndexStream stream = children.poll();
+            
+            // If the top of the shard-sorted priority queue is beyond the specified shard then we can bail out.
+            String streamDayOrShard = stream.peek().first();
+            
+            // Avoid seeking if the stream shard matches the seek shard, or if the stream shard is greater than the seek shard.
+            if (ShardEquality.matches(streamDayOrShard, seekShard) || ShardEquality.greaterThanOrEqual(streamDayOrShard, seekShard)) {
+                nextChildren.add(stream);
+                continue;
+            }
```

Comment:
is there a guarantee this will always be the lowest?

Reply:
The PeekOrdering used on the queue sorts by shard, so this will always pull the lowest.